ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+    arcv_hr_start = get_obs_arcv_hr(obtype, arcv_intvl_hrs, hod_first)
+
+    # Ending archive hour.  This is set to the archive hour containing obs at
+    # the last obs retrieval time of the day.
+    hod_last = obs_retrieve_times_crnt_day[-1].hour
+    arcv_hr_end = get_obs_arcv_hr(obtype, arcv_intvl_hrs, hod_last)
+
+    # Set other variables needed below when evaluating the METplus template for
+    # the full path to the processed observation files.
+    one_hour = dt.timedelta(hours=1)
+    ushdir = config['user']['USHdir']
+
+    # Create dictionary containing the paths to all the processed obs files
+    # that should exist once this script successfully completes.  In this
+    # dictionary, the keys are the field groups, and the values are lists of
+    # paths.  Here, by "paths to processed files" we mean the paths after any
+    # renaming and rearrangement of files that this script may do to the "raw"
+    # files, i.e. the files as they are named and arranged within the archive
+    # (tar) files on HPSS.
+    all_fp_proc_dict = {}
+    for fg, fp_proc_templ in zip(field_groups_in_obs, fp_proc_templates):
+        all_fp_proc_dict[fg] = []
+        for yyyymmddhh in obs_retrieve_times_crnt_day:
+            # Set the lead hour, i.e. the number of hours from the beginning of the
+            # day at which the file is valid.
+            lhr = int((yyyymmddhh - yyyymmdd_task)/one_hour)
+            # Call a bash script to evaluate the template for the full path to the
+            # file containing METplus timestrings at the current time.  This should
+            # be upgraded to a python script at some point.
+            cmd = '; '.join(['export USHdir=' + ushdir,
+                             'export yyyymmdd_task=' + yyyymmdd_task_str,
+                             'export lhr=' + str(lhr),
+                             'export METplus_timestr_tmpl=' + fp_proc_templ,
+                              os.path.join(ushdir, 'run_eval_METplus_timestr_tmpl.sh')])
+            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
+            fp_proc = result.stdout.strip()
+            all_fp_proc_dict[fg].append(fp_proc)
+
+    # Check whether any obs files already exist on disk in their processed
+    # (i.e. final) locations.  If so, adjust the starting archive hour.  In
+    # the process, keep a count of the number of obs files that already exist
+    # on disk.
+    num_existing_files = 0
+    do_break = False
+    for fg in field_groups_in_obs:
+        for yyyymmddhh, fp_proc in zip(obs_retrieve_times_crnt_day, all_fp_proc_dict[fg]):
+            # Check whether the processed file already exists.
+            if os.path.isfile(fp_proc):
+                num_existing_files += 1
+                msg = dedent(f"""
+                    File already exists on disk:
+                        fp_proc = {fp_proc}
+                    """)
+                logging.info(msg)
```

Comment:
Low priority message
```suggestion
                logging.debug(msg)
```

Reply:
Changed.